src/codecs/jpeg/entropy.rs

```diff
@@ -1,9 +1,9 @@
 /// Given an array containing the number of codes of each code length,
 /// this function generates the huffman codes lengths and their respective
 /// code lengths as specified by the JPEG spec.
-fn derive_codes_and_sizes(bits: &[u8]) -> (Vec<u8>, Vec<u16>) {
-    let mut huffsize = vec![0u8; 256];
-    let mut huffcode = vec![0u16; 256];
+fn derive_codes_and_sizes(bits: &[u8]) -> ([u8; 256], [u16; 256]) {
+    let mut huffsize = [0u8; 256];
+    let mut huffcode = [0u16; 256];
```

Comment:
Big arrays on the stack make me slightly nervous, but this is a leaf function and 768 bytes isn't that big so this is probably fine.